Source/WebCore/platform/graphics/gstreamer/eme/WebKitCommonEncryptionDecryptorGStreamer.cpp

```diff
@@ -241,9 +242,14 @@ static GstFlowReturn webkitMediaCommonEncryptionDecryptTransformInPlace(GstBaseT
             GST_ERROR_OBJECT(self, "can't process key requests in less than PAUSED state");
             return GST_FLOW_NOT_SUPPORTED;
         }
-        if (!priv->m_condition.waitFor(priv->m_mutex, WEBCORE_GSTREAMER_EME_LICENSE_KEY_RESPONSE_TIMEOUT, [priv] { return priv->m_keyReceived; })) {
-            GST_ERROR_OBJECT(self, "key not available");
-            return GST_FLOW_NOT_SUPPORTED;
+        if (!priv->m_condition.waitFor(priv->m_mutex, WEBCORE_GSTREAMER_EME_LICENSE_KEY_RESPONSE_TIMEOUT, [priv] { return priv->m_keyReceived || priv->m_isFlushing; })) {
+            if (priv->m_isFlushing) {
+                GST_DEBUG_OBJECT(self, "flushing");
+                return GST_FLOW_FLUSHING;
+            } else {
+                GST_ERROR_OBJECT(self, "key not available");
+                return GST_FLOW_FLUSHING;
```

Comment:
Shouldn't it be `GST_FLOW_ERROR` in this line? You are not flushing in that case, but you have discovered that you can't continue decoding (because you don't have a key)

Reply:
Yes, copy&paste error. It's a pity because my env is broken so I couldn't test this. It should be fixed now.